ta/ta.mk

```diff
@@ -90,26 +91,40 @@ $$(foreach h, $$(sf), $$(eval $$(call copy-file, $1/$$(h), \
 	$$(patsubst %/,%,$$(subst /./,/,$2/$$(dir $$(h)))))))
 endef
 $(foreach d, $(incdirs$(sm)), \
-	$(eval $(call copy-incdir, $(d), $(out-dir)/export-user_ta/include)))
+	$(eval $(call copy-incdir, $(d), $(out-dir)/export-$(sm)/include)))
 
 # Copy the .h files needed by host
 $(foreach d, $(incdirs-host), \
-	$(eval $(call copy-incdir, $(d), $(out-dir)/export-user_ta/host_include)))
+	$(eval $(call copy-incdir, $(d), $(out-dir)/export-$(sm)/host_include)))
 $(foreach f, $(incfiles-extra-host), \
-	$(eval $(call copy-file, $(f), $(out-dir)/export-user_ta/host_include)))
+	$(eval $(call copy-file, $(f), $(out-dir)/export-$(sm)/host_include)))
 
 # Copy the src files
 ta-srcfiles = ta/arch/$(ARCH)/user_ta_header.c \
-	$(wildcard ta/arch/$(ARCH)/user_ta_elf_arm.lds)
+	$(wildcard ta/arch/$(ARCH)/ta.ld.S)
 $(foreach f, $(ta-srcfiles), \
-	$(eval $(call copy-file, $(f), $(out-dir)/export-user_ta/src)))
+	$(eval $(call copy-file, $(f), $(out-dir)/export-$(sm)/src)))
 
 # Copy keys
 ta-keys = keys/default_ta.pem
 $(foreach f, $(ta-keys), \
-	$(eval $(call copy-file, $(f), $(out-dir)/export-user_ta/keys)))
+	$(eval $(call copy-file, $(f), $(out-dir)/export-$(sm)/keys)))
 
 # Copy the scripts
 ta-scripts = $(wildcard scripts/sign.py)
 $(foreach f, $(ta-scripts), \
-	$(eval $(call copy-file, $(f), $(out-dir)/export-user_ta/scripts)))
+	$(eval $(call copy-file, $(f), $(out-dir)/export-$(sm)/scripts)))
+
+# Create config file
+conf-file := $(out-dir)/export-$(sm)/mk/conf.mk
+sm-$(conf-file) := $(sm)
+.PHONY: $(conf-file)
+$(conf-file):
+	@echo '  CREATE ' $@
+	@echo sm := $(sm-$(@)) > $@
```

Comment:
You should use `@$(cmd-echo-silent) ` for the `CREATE` message (which should maybe be `GEN` for consistency?) and `$(q)` for the actual commands.